From 0252aa08aafb4a40ea2d821f58e88e99a644b097 Mon Sep 17 00:00:00 2001
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Mon, 29 Mar 2021 15:45:24 +0800
Subject: [PATCH] sdei_watchdog: Fix compile error when PPC_WATCHDOG is disable
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 0252aa08aafb4a40ea2d821f58e88e99a644b097

 on PowerPC
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

hulk inclusion
category: bugfix
bugzilla: 48046
CVE: NA

---------------------------------

When I compile the kernel with CONFIG_PPC_WATCHDOG is disabled on
PowerPC, I got the following compile error:
 In file included from kernel/hung_task.c:11:0:
 ./include/linux/nmi.h: In function ‘touch_nmi_watchdog’:
 ./include/linux/nmi.h:143:2: error: implicit declaration of function ‘arch_touch_nmi_watchdog’; did you mean ‘touch_nmi_watchdog’? [-Werror=implicit-function-declaration]
   arch_touch_nmi_watchdog();
   ^~~~~~~~~~~~~~~~~~~~~~~
   touch_nmi_watchdog

It is because CONFIG_HARDLOCKUP_DETECTOR_PERF is still enabled in my
situation. Fix it by excluding arch_touch_nmi_watchdog() only when
CONFIG_PPC_WATCHDOG is disabled.

Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/nmi.h   | 2 +-
 kernel/watchdog_hld.c | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/linux/nmi.h b/include/linux/nmi.h
index 5a4ce2eca99e..31abe433345f 100644
--- a/include/linux/nmi.h
+++ b/include/linux/nmi.h
@@ -96,7 +96,7 @@ static inline void hardlockup_detector_disable(void) {}
 #endif
 
 #if defined(CONFIG_HARDLOCKUP_DETECTOR)
-#ifndef CONFIG_PPC
+#ifndef CONFIG_PPC_WATCHDOG
 extern void arch_touch_nmi_watchdog(void);
 #endif
 #else
diff --git a/kernel/watchdog_hld.c b/kernel/watchdog_hld.c
index a3f35067b4d0..ce26950a8140 100644
--- a/kernel/watchdog_hld.c
+++ b/kernel/watchdog_hld.c
@@ -26,8 +26,8 @@ static DEFINE_PER_CPU(bool, watchdog_nmi_touch);
 
 static unsigned long hardlockup_allcpu_dumped;
 
-#ifndef CONFIG_PPC
-notrace void __weak arch_touch_nmi_watchdog(void)
+#ifndef CONFIG_PPC_WATCHDOG
+notrace void arch_touch_nmi_watchdog(void)
 {
 	/*
 	 * Using __raw here because some code paths have
-- 
2.26.2

